Core.ApplicationServices/Organizations/OrganizationUnitService.cs

```diff
@@ -0,0 +1,345 @@
+        {
+            return GetOrganization(id, accessLevel)
+                .Match
+                (
+                    organization => _authorizationContext.AllowModify(organization) == false 
+                        ? new OperationError("User is not allowed to modify the organization", OperationFailure.Forbidden) 
+                        : Maybe<OperationError>.None,
+                    error => error
+                );
+        }
+
+        private Maybe<OperationError> RemovePayments(IEnumerable<PaymentChangeParameters> payments)
+        {
+            foreach (var payment in payments)
+            {
+                var removeInternalPaymentsResult = _contractService.RemovePaymentResponsibleUnits(payment.ItContractId, true, payment.InternalPayments);
+                if (removeInternalPaymentsResult.HasValue)
+                    return removeInternalPaymentsResult.Value;
+
+                var removeExternalPaymentsResult = _contractService.RemovePaymentResponsibleUnits(payment.ItContractId, false, payment.ExternalPayments);
+                if (removeExternalPaymentsResult.HasValue)
+                    return removeExternalPaymentsResult.Value;
+            }
+
+            return Maybe<OperationError>.None;
+        }
+
+        private Maybe<OperationError> RemoveContractRegistrations(IEnumerable<int> contractIds)
+        {
+            foreach (var contractId in contractIds)
+            {
+                var deleteResult = _contractService.RemoveContractResponsibleUnit(contractId);
+                if (deleteResult.HasValue)
+                    return deleteResult.Value;
+            }
+
+            return Maybe<OperationError>.None;
+        }
+
+        private Maybe<OperationError> RemoveSystemRelevantUnits(IEnumerable<int> systemIds, int unitId)
+        {
+            foreach (var systemId in systemIds)
+            {
+                var deleteResult = _usageService.RemoveRelevantUnit(systemId, unitId);
+                if (deleteResult.HasValue)
+                    return deleteResult.Value;
+            }
+
+            return Maybe<OperationError>.None;
+        }
+
+        private Maybe<OperationError> RemoveSystemResponsibleRegistrations(IEnumerable<int> systemIds)
+        {
+            foreach (var systemId in systemIds)
+            {
+                var deleteResult = _usageService.RemoveResponsibleUsage(systemId);
+                if (deleteResult.HasValue)
+                    return deleteResult.Value;
+            }
+
+            return Maybe<OperationError>.None;
+        }
+
+        private Maybe<OperationError> TransferPayments(int targetUnitId, IEnumerable<PaymentChangeParameters> payments)
+        {
+            foreach (var payment in payments)
+            {
+                var transferInternalPaymentsResult = _contractService.TransferPayments(payment.ItContractId, targetUnitId, true, payment.InternalPayments);
```

Comment:
remember all of theste `Maybe<OperationError>` results should not be suffixed with "result" but with "Error"... that makes it easier to understand why the code is bailing out on "HasValue"